### PR TITLE
Added Decorators

### PR DESCRIPTION
**Pull request by AbelG101**
## Changes
- Added a Nameable class
- Added base decorator class:
    - with a method correct_name
- Added 2 concrete decorators:
    - CapitalizeDecorator with correct_name method to capitalize the name
    - TrimmerDecorator with correct_name method to trim names with characters longer than 10.

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'method should be implemented in concrete class'
```
Comment (mikethreels) [inline nameable.rb]: - [ ] [OPTIONAL] good job adding the error message, you might even want to use string interpolation here to make the message interactive like below:
```
"#{self.class} has not implemented method '#{__method__}'"
```

File: base_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
```
Comment (mikethreels) [inline base_decorator.rb]: - [ ] You are required to add the attribute accessor for `@nameable` in this class

File: trimmer_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require_relative './base_decorator'
+require_relative './person'
+require_relative './capitalize_decorator'
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    return @nameable.correct_name[0, 10] if @nameable.correct_name.length > 10
```
Comment (mikethreels) [inline trimmer_decorator.rb]: - [ ] if you want the name to have a max length of 10 characters it should look like `return @nameable.correct_name[0, 9] if @nameable.correct_name.length > 10`

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Abel, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mateo951 — APPROVED** [review]
### Hi @AbelG101, 

Well done making the changes requested by the previous reviewer 😎
Your project is complete!  There is nothing else to say other than... it's time to merge it 🐐 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**